### PR TITLE
Support Azure Database

### PR DESCRIPTION
**Pull request by mw2q**
Generalize the RDS specific paths as DBaaS since it can be shared with
Azure Database.

Azure Database has some unique usage with respect to how the username
needs to be generated in a connect strict as opposed to generate usage
in SQL statements.  HammerDB as well as some of the scripts needs to be
generated differently because of this.  There is currently an open
ticket with HammerDB to handle this.  This change addresses Azure
Database directly for ease of development.

### CODE REVIEW

File: roles/setup_hammerdb/templates/loader.tcl.template [hunk 1]
```diff
@@ -11,6 +11,12 @@ diset connection pg_port {{ pg_port }}
 diset tpcc pg_count_ware {{ tprocc_warehouses }}
 diset tpcc pg_num_vu {{ ansible_processor_vcpus }}
 diset tpcc pg_raiseerror true
-diset tpcc pg_superuser {{ pg_superuser }}
+{% if azure_db_hackery %}
```
Comment (jt-edb) [inline roles/setup_hammerdb/templates/loader.tcl.template:14]: This variable does not look to be defined in `setup_hammerdb/defaults/main.yml`

File: roles/setup_hammerdbserver/tasks/install_hammerdb.yml [hunk 2]
```diff
@@ -7,3 +7,22 @@
...
+# We need this Azure Database hack until this is resolved:
+# https://github.com/TPC-Council/HammerDB/issues/163
+
+- name: Create HammerDB patch for Azure Database
+  template:
+    dest: hammerdb-{{ hammerdb_version }}-azure-db.patch
+    src: hammerdb-{{ hammerdb_version }}-azure-db.patch.template
+    mode: 0644
+  become: no
+  when: azure_db_hackery is defined
```
Comment (jt-edb) [inline roles/setup_hammerdbserver/tasks/install_hammerdb.yml:20]: I'd suggest to check if `azure_db_hackery` is `True`

**Review by jt-edb — COMMENTED** [review]
Thank you

File: roles/setup_hammerdb/templates/loader.tcl.template (hunk 1 above)
Comment (mw2q) [reply roles/setup_hammerdb/templates/loader.tcl.template:14]: Right, it's set on the fly when a `deploy` is executed: https://github.com/mw2q/postgres-deployment/blob/azure-db/edbdeploy/project.py#L1261

File: roles/setup_hammerdbserver/tasks/install_hammerdb.yml (hunk 2 above)
Comment (mw2q) [reply roles/setup_hammerdbserver/tasks/install_hammerdb.yml:20]: Done,

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**